### ISSUE TITLE
Reflection: getDefaultValue of multi fails

### ISSUE THREAD

**Issue by ostrolucky** [phpredis/phpredis#2044]
### Expected behaviour
It should return `1`, as specified at  https://github.com/phpredis/phpredis/blob/eb44ea9aa453192f64a24da0972c9448e015cddb/redis.stub.php#L273

### Actual behaviour
```
ReflectionException with message 'Internal error: Failed to retrieve the default value'
```

### I'm seeing this behaviour on
- OS: MacOS 11.6.1
- PHP: 8.1.0
- phpredis: 5.3.4

### Steps to reproduce, backtrace or example script
```
>>> (new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue()
ReflectionException with message 'Internal error: Failed to retrieve the default value'
```

### I've checked
- [x] There is no similar issue from other users
- [x] Issue isn't fixed in `develop` branch

This is related to https://github.com/snc/SncRedisBundle/issues/640. This issue in phpredis causes generating wrong proxy of `multi` function (and possibly others)

**Comment by yatsukhnenko** [comment]
@ostrolucky shouldn't `ReflectionParameter::getDefaultValueConstantName` be used in this case?

**Comment by ostrolucky** [comment]
I don't think so, we are interested in default value, so that proxy is able to pass it as default value, not constant name. However, neither works
```
>>> (new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValueConstantName()
ReflectionException with message 'Internal error: Failed to retrieve the default value'
```

**Comment by yatsukhnenko** [comment]
@ostrolucky It might be problem in testing environment
```
yatsukhnenko@localhost~/code/phpredis (develop)$ php -v
PHP 8.1.0 (cli) (built: Dec  8 2021 09:04:03) (NTS DEBUG)
Copyright (c) The PHP Group
Zend Engine v4.1.0, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.0, Copyright (c), by Zend Technologies
```

```
yatsukhnenko@localhost~/code/phpredis (develop)$ php -d extension=modules/redis.so -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());"
int(1)
```

```
yatsukhnenko@localhost~/code/phpredis (develop)$ php -d extension=modules/redis.so -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValueConstantName());"
string(12) "Redis::MULTI"
```

**Comment by noofaq** [comment]
@yatsukhnenko In fact I have originated the base issue, so I have just tested on my local environment and I receive same error as @ostrolucky 
![obraz](https://user-images.githubusercontent.com/1412384/146201578-c46b4556-6c26-4273-be3e-ac0d6c1594c3.png)

My configuration is  running in docker container based on `Debian11` using `PHP 8.0.13` and `php redis 5.3.4` (all packages are coming from sury PPA - `https://packages.sury.org/php/` )

**Comment by yatsukhnenko** [comment]
Could you try to test this code on the same environment but different extension?

**Comment by noofaq** [comment]
@yatsukhnenko Could you explain what do you mean?

I have just perform very basic trial: started container with official php docker image (`php/cli-bullseye` to be precise) which bases on PHP 8.1.0 and installed there redis extension using pecl (basic `pecl install redis`). 
This allowed me to reproduce the issue very quickly.

Just to add: same error happens on the latest non-stable version `5.3.5RC1` (previous stable version `5.3.3` fails too)

**Comment by yatsukhnenko** [comment]
I'm asking to run something like `php -r "var_dump((new ReflectionClass('ReflectionClass'))->getMethod('getMethod')->getParameters()[0]->getDefaultValue());"` on your environment

**Comment by noofaq** [comment]
Thank you for the explanation.

The exact function call you asked has failed in the same way:
![obraz](https://user-images.githubusercontent.com/1412384/146208578-e80dc379-98f2-44e8-9562-21d2659c4286.png)

But method ReflectionClass::getMethod does not have any default value for the first parameter.

Trying different method ReflectionClass::getAttributes provides correct result:
![obraz](https://user-images.githubusercontent.com/1412384/146208416-1abdfe17-5a57-4060-910c-9b080f050c3e.png)

**Comment by ostrolucky** [comment]
```
✔ /vagrant/fe-bl-service [develop ↓·2|✚ 2…6⚑ 2]
16:29 $ php -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());"
PHP Fatal error:  Uncaught ReflectionException: Internal error: Failed to retrieve the default value in Command line code:1
Stack trace:
#0 Command line code(1): ReflectionParameter->getDefaultValue()
#1 {main}
  thrown in Command line code on line 1

Fatal error: Uncaught ReflectionException: Internal error: Failed to retrieve the default value in Command line code on line 1

ReflectionException: Internal error: Failed to retrieve the default value in Command line code on line 1

Call Stack:
    0.0001     392256   1. {main}() Command line code:0
    0.0001     392408   2. ReflectionParameter->getDefaultValue() Command line code:1

✘-255 /vagrant/fe-bl-service [develop ↓·2|✚ 2…6⚑ 2]
16:29 $ php --version
PHP 8.0.12 (cli) (built: Oct 22 2021 12:37:40) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.12, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.12, Copyright (c), by Zend Technologies
    with Xdebug v3.1.1, Copyright (c) 2002-2021, by Derick Rethans
✔ /vagrant/fe-bl-service [develop ↓·2|✚ 2…6⚑ 2]
16:29 $ php -r "var_dump(phpversion('redis'));"
Command line code:1:
string(5) "5.3.4"
```

**Comment by ostrolucky** [comment]
Looks like this was indeed somehow solved in develop

This reproduces issue:

```
➜  repro ls -ls
total 8
8 -rw-r--r--  1 gabriel.ostrolucky  OFFICE\Domänen-Benutzer  187 Dec 16 22:33 Dockerfile
➜  repro cat Dockerfile
FROM php:8.1-cli

RUN pecl install redis
RUN docker-php-ext-enable redis
RUN php -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());"
➜  repro docker build .
[+] Building 0.9s (7/7) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                             0.0s
 => => transferring dockerfile: 43B                                                                                                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/php:8.1-cli                                                                                                                                                                                   0.6s
 => [1/4] FROM docker.io/library/php:8.1-cli@sha256:b2afd03bcad73aa0a207b0be83310fb2cb35b7a47aceb89e82437b1d19bf81c5                                                                                                                             0.0s
 => CACHED [2/4] RUN pecl install redis                                                                                                                                                                                                          0.0s
 => CACHED [3/4] RUN docker-php-ext-enable redis                                                                                                                                                                                                 0.0s
 => ERROR [4/4] RUN php -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());"                                                                                                                0.2s
------
 > [4/4] RUN php -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());":
#7 0.211
#7 0.211 Fatal error: Uncaught ReflectionException: Internal error: Failed to retrieve the default value in Command line code:1
#7 0.211 Stack trace:
#7 0.211 #0 Command line code(1): ReflectionParameter->getDefaultValue()
#7 0.211 #1 {main}
#7 0.211   thrown in Command line code on line 1
------
executor failed running [/bin/sh -c php -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());"]: exit code: 255
```

But this doesn't:
```
➜  repro ls
Dockerfile
➜  repro cat Dockerfile
FROM php:8.1-cli

RUN apt-get update && apt-get install -y git

RUN git clone https://github.com/phpredis/phpredis.git
WORKDIR phpredis
RUN phpize
RUN ./configure
RUN make && make install

RUN docker-php-ext-enable redis
RUN php -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());"
➜  repro docker build .
[+] Building 0.7s (13/13) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                             0.0s
 => => transferring dockerfile: 43B                                                                                                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/php:8.1-cli                                                                                                                                                                                   0.6s
 => [1/9] FROM docker.io/library/php:8.1-cli@sha256:b2afd03bcad73aa0a207b0be83310fb2cb35b7a47aceb89e82437b1d19bf81c5                                                                                                                             0.0s
 => CACHED [2/9] RUN apt-get update && apt-get install -y git                                                                                                                                                                                    0.0s
 => CACHED [3/9] RUN git clone https://github.com/phpredis/phpredis.git                                                                                                                                                                          0.0s
 => CACHED [4/9] WORKDIR phpredis                                                                                                                                                                                                                0.0s
 => CACHED [5/9] RUN phpize                                                                                                                                                                                                                      0.0s
 => CACHED [6/9] RUN ./configure                                                                                                                                                                                                                 0.0s
 => CACHED [7/9] RUN make && make install                                                                                                                                                                                                        0.0s
 => CACHED [8/9] RUN docker-php-ext-enable redis                                                                                                                                                                                                 0.0s
 => CACHED [9/9] RUN php -r "var_dump((new ReflectionClass('Redis'))->getMethod('multi')->getParameters()[0]->getDefaultValue());"                                                                                                               0.0s
 => exporting to image                                                                                                                                                                                                                           0.0s
 => => exporting layers                                                                                                                                                                                                                          0.0s
 => => writing image sha256:d20b813af1672fc7aced09e9f88453301d955090798adb5e59912eae05ca6d5c                                                                                                                                                     0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
```

**Comment by ostrolucky** [comment]
5.3.5 was released since then, but somehow doesn't fix the issue. Any idea why @yatsukhnenko this bug is in pecl version but not when compiled manually?

**Comment by yatsukhnenko** [comment]
@ostrolucky 5.3.5 doesn't contain arginfo based on stub files. We are going to release it in 6.0.0 because it's a breaking change

**Comment by michael-grunder** [comment]
Closing as the problem has been fixed in the `develop` branch and will be released with PhpRedis 6.0.0
